src/vmm/src/arch/aarch64/regs.rs

```diff
@@ -195,207 +195,224 @@ pub fn get_manufacturer_id_from_host() -> Result<u32, Error> {
     Ok(manufacturer_id >> 24)
 }
 
-/// Configure relevant boot registers for a given vCPU.
-///
-/// # Arguments
-///
-/// * `vcpu` - Structure for the VCPU that holds the VCPU's fd.
-/// * `cpu_id` - Index of current vcpu.
-/// * `boot_ip` - Starting instruction pointer.
-/// * `mem` - Reserved DRAM for current VM.
-pub fn setup_boot_regs(
-    vcpu: &VcpuFd,
-    cpu_id: u8,
-    boot_ip: u64,
-    mem: &GuestMemoryMmap,
-) -> Result<(), Error> {
-    let kreg_off = offset__of!(kvm_regs, regs);
-
-    // Get the register index of the PSTATE (Processor State) register.
-    let pstate = offset__of!(user_pt_regs, pstate) + kreg_off;
-    let id = arm64_core_reg_id!(KVM_REG_SIZE_U64, pstate);
-    vcpu.set_one_reg(id, PSTATE_FAULT_BITS_64.into())
-        .map_err(|err| Error::SetOneReg(id, err))?;
-
-    // Other vCPUs are powered off initially awaiting PSCI wakeup.
-    if cpu_id == 0 {
-        // Setting the PC (Processor Counter) to the current program address (kernel address).
-        let pc = offset__of!(user_pt_regs, pc) + kreg_off;
-        let id = arm64_core_reg_id!(KVM_REG_SIZE_U64, pc);
-        vcpu.set_one_reg(id, boot_ip.into())
-            .map_err(|err| Error::SetOneReg(id, err))?;
+/// Trait with additional methods for VcpuFd
+pub trait Aarch64VcpuUtils {
```

Comment:
what's the benefit for having a trait here?

Reply:
It is to extend methods of `VcpuFd`. I prefer this way over free standing functions. This way if you want to use these methods you just import the trait instead of bunch of methods.

Reply:
I disagree with the approach of using a trait here, with traits there are additional complexities and restrictions that are best avoided. Outside exceptional circumstances I think traits should only be used when they are exported.

Reply:
I would agree with Jonathan. `*Utils` doesn't look like a good trait, because it's likely to change over time. What about isolating these functions in a module instead? They can be imported altogether easily then too.

Reply:
Alright, if you are really against this trait I can remove it and revert back to the free standing functions. At least this will save ~60 loc. 
I still prefer this:
```rust
use arch::aarch64::Aarch64VcpuUtils;

vcpu.get_all_registers(&mut regs);
vcpu.set_registers(&regs);
```
over this
```rust
use arch::aarch64::{set_registers, get_all_registers};

get_all_registers(&vcpu, &mut regs);
set_registers(&vcpu, &regs);
```
but ultimately it does not matter.
@JonathanWoollett-Light @kalyazin will this be ok with you?